### PR TITLE
Feature/expandable schemas

### LINKED ISSUE #140: Access schema include paths
Need to add the ability to get the actual !include path for schemas.

### PR DESCRIPTION
**Pull request by alvassin**
Added flag `dereferenceSchemas`, that allows to preprocess json-schemas and provide resolved $refs. Fixes #140

### CODE REVIEW

File: src/raml.coffee [hunk 1]
```diff
@@ -171,14 +172,39 @@ class @RamlParser
...
+        deferred.reject(err)
+      )
+    return deferred.promise
+
+  checkPendingFileIsSchema: (loader, fileInfo, node) ->
+    if fileInfo.parentKey and fileInfo.parentKey.value is 'schema'
+      return true
+
+    if fileInfo.parentNode
+      schemasSeqNode = loader.findSchemasSequenceNode(node)
```
Comment (dmartinezg) [inline src/raml.coffee]: What if you have this RAML:

```raml
#%RAML 0.8
/foo:
  post:
    body:
      schema: !include someFastFile.json
schemas: !include someSlowFile.yaml
```

You should defer all schema reading and processing, until all !included files have been loaded.

Comment (alvassin) [inline src/raml.coffee]: @dmartinezg got it. thanks!

Comment (alvassin) [inline src/raml.coffee]: @dmartinezg can you please give an example how `someSlowFile.yaml` should look like?

Comment (dmartinezg) [inline src/raml.coffee]: Here is a sample test, with random response times for includes:

https://github.com/raml-org/raml-js-parser/blob/master/test/specs/regressions.js#L960

**Comment by alvassin** [comment]
@dmartinezg I implemented support for all `!include`-d schemas, they work correctly now. 

As for inline-specified schemes, that have $refs (see below), I need [this issue](https://github.com/BigstickCarpet/json-schema-ref-parser/issues/13) to be fixed. 

After i will get some response from maintainers i will add support for such schemas too.
```yaml
... raml definition ...
responses:
  200:
    body:
      application/json:
        schema: |
          {
            "$schema": "http://json-schema.org/draft-04/schema#",
            "type": "object",
            "properties": {
              "items": {
                "$ref": "./schemas/collections/items.json"
              }
            }
          }
```

**Comment by alvassin** [comment]
@dmartinezg please check code & tests, all cases should be covered now.

**Comment by dmartinezg** [comment]
@alvassin, apologies for reviewing late, I've had a big mail backlog for the past few days.

I think it is ready to merge, have you signed this?: https://api-notebook.anypoint.mulesoft.com/notebooks#bc1cf75a0284268407e4

After you do that, I can merge, tag and deploy to npm

**Comment by alvassin** [comment]
@dmartinezg you've got nice automation. Signed.

**Comment by dmartinezg** [comment]
Oh, please, squash commits

**Comment by alvassin** [comment]
@dmartinezg done.

**Comment by dmartinezg** [comment]
@alvassin, thanks!

**Pull request merged**